### PR TITLE
Shellcheck rewrites

### PR DESCRIPTION
**Pull request by mehrdadn**
## Why are these changes needed?

Resolves shellcheck issues that may require rewriting code or making semantic changes.

While independent, it may be better to merge this after #9596.

## Related issue number

#8574

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: doc/examples/lm/ray_train.sh [hunk 1]
```diff
@@ -4,7 +4,6 @@ TOTAL_UPDATES=125000       # Total number of training steps
 WARMUP_UPDATES=10000       # Warmup the learning rate over this many updates
 PEAK_LR=0.0005             # Peak learning rate, adjust as needed
 TOKENS_PER_SAMPLE=512      # Max sequence length
-MAX_POSITIONS=512          # Num. positional embeddings (usually same as above)
```
Comment (edoakes) [inline doc/examples/lm/ray_train.sh:7]: Is this unused? Just making sure the change is intentional (same as below).

**Review by edoakes — APPROVED** [review]
LGTM just one comment to make sure it isn't an errant change

File: doc/examples/lm/ray_train.sh (hunk 1 above)
Comment (mehrdadn) [reply doc/examples/lm/ray_train.sh:7]: Yup this was intentional, they do appear to be unused.

Comment (mehrdadn) [reply doc/examples/lm/ray_train.sh:7]: Also I merged in master but CI seems clogged, here's the build on my fork: https://github.com/mehrdadn/ray/runs/906384304?check_suite_focus=true

**Pull request merged**